client/client.go

```diff
@@ -289,20 +401,26 @@ func (client *Client) request() *request {
 	return client.requestPool.Get().(*request)
 }
 
-func (client *Client) submit(pt rt.PT, funcname string, payload []byte) (string, error) {
-	var err error
+func (client *Client) submit(pt rt.PT, funcname string, payload []byte) (handle string, err error) {
 
-	client.outbound <- client.request().submitJob(pt, funcname, IdGen.Id(), payload)
+	defer func() {
```

Comment:
This function is now present at least 3 times. Can we refactor it?

Reply:
Good idea, I looked around but it looks like it's the shortest form so far I can think of where (a) we are deferring a function (b) which calls recover(), (c) assigns named return variables (e.g. handle and err in this case), and (d) one of the return variables default values differs between "" (emptry string), or nil, but let me know if you've seen a shorter construct though.